presto-pinot-toolkit/src/test/java/com/facebook/presto/pinot/query/TestPinotPlanOptimizerSql.java

```diff
@@ -30,4 +39,28 @@ public void assertUsingSqlSyntax()
     {
         Assert.assertEquals(defaultSessionHolder.getConnectorSession().getProperty("use_pinot_sql_for_broker_queries", Boolean.class).booleanValue(), true);
     }
+
+    @Test
+    public void testDistinctLimitPushdown()
+    {
+        PlanBuilder planBuilder = createPlanBuilder(defaultSessionHolder);
+        PlanNode originalPlan = distinctLimit(
```

Comment:
I didn't follow why distinct limit generates the same PQL as a regular limit ?

Reply:
It's adding group by

Reply:
basically convert `SELECT DISTINCT col FROM myTable LIMIT 10` to `SELECT col FROM myTable GROUP BY col LIMIT 10`